sklearn/decomposition/_nmf.py

```diff
@@ -781,13 +873,36 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
     n_iter : int
         The number of iterations done by the algorithm.
 
+    iter_offset : int
+        The number of iteration on data batches that has been
+        performed.
+
     References
     ----------
+    Lee, D. D., & Seung, H., S. (2001). Algorithms for Non-negative Matrix
+    Factorization. Adv. Neural Inform. Process. Syst.. 13.
     Fevotte, C., & Idier, J. (2011). Algorithms for nonnegative matrix
     factorization with the beta-divergence. Neural Computation, 23(9).
     """
     start_time = time.time()
 
+    n_samples = X.shape[0]
+
+    A = None
+    B = None
+
+    if batch_size is None:
+        batch_size = n_samples
+
+    if batch_size < n_samples:
+        # Initialize auxiliary matrices
+        A = H.copy()
+        B = np.ones(H.shape)
+
+    rho = 0.
+    if forget_factor is not None:
+        rho = forget_factor ** (batch_size / n_samples)
```

Comment:
I know it's how it's defined in the paper but I find that  ``** (batch_size / n_samples)`` makes the forget_factor very hard to set. When the ratio is small, ``rho`` gets close to one no matter how low you set the forget_factor. For instance if `n_samples = 100000` and `batch_size = 256` I have to set `forget_factor = 1e-18` to have `rho <  0.9`.